### PR TITLE
Synapse Spark2.4/JDBC 8.4.1 branch

### PR DESCRIPTION
**Pull request by shivsood**
- JDBC 8.4.1
- Removed deprecated interfaces

### CODE REVIEW

**Comment by shivsood** [comment]
Panjak confirmed this worked on Synapse.

**Comment by shivsood** [comment]
@luxu1-ms @shanyu @rajmera3 can u please review? Plan to merge this to master to enable synapse.

**Comment by rajmera3** [comment]
@shivsood Do you mind updating the readme in this PR to include 8.4 as a compatible JDBC version? Thanks

**Comment by shivsood** [comment]
> @shivsood Do you mind updating the readme in this PR to include 8.4 as a compatible JDBC version? Thanks
Would do that. Thanks.

**Review by rajmera3 — APPROVED** [review]

File: src/main/java/com/microsoft/sqlserver/jdbc/spark/DataFrameBulkRecord.java [hunk 1]
```diff
@@ -82,41 +78,11 @@ public int getScale(int column) {
         return dfColumnMetadata[column-1].getScale();
     }
 
-    @Override
-    public boolean isAutoIncrement(int column) {
```
Comment (luxu1-ms) [inline src/main/java/com/microsoft/sqlserver/jdbc/spark/DataFrameBulkRecord.java:86]: @shivsood do we also need to remove the field isAutoIncrement and the function isAutoIncrement() in ColumnMetadata.java?

**Review by luxu1-ms — APPROVED** [review]

**Comment by shivsood** [comment]
Validated on BDC with 8.4.1. All test cases passed. Wont work with 7.0, error because of following interface change.

An error occurred while calling o92.save.
: java.lang.NoClassDefFoundError: com/microsoft/sqlserver/jdbc/ISQLServerBulkData
	at com.microsoft.sqlserver.jdbc.spark.DefaultSource.createRelation(DefaultSource.sca

File: src/main/java/com/microsoft/sqlserver/jdbc/spark/DataFrameBulkRecord.java [hunk 2]
```diff
@@ -14,29 +14,23 @@
 package com.microsoft.sqlserver.jdbc.spark;
 
 import java.util.Set;
-import java.util.Collections;
 import java.util.stream.Collectors;
 import java.util.stream.IntStream;
-import java.time.format.DateTimeFormatter;
```
Comment (shanyu) [inline src/main/java/com/microsoft/sqlserver/jdbc/spark/DataFrameBulkRecord.java:20]: These are previously not used imports, right?

**Review by shanyu — APPROVED** [review]

**Pull request merged**